### PR TITLE
Better memory and storage info

### PR DESCRIPTION
**Pull request by vjr**
Stolen code from GNOME settings repo to use UDev and UDisks libraries to get more accurate memory and storage info.

Memory now shows the physical RAM capacity and storage code collects the total size of all non-removable and non-ejectable disks.

Before: (note the non-physical RAM size using glibtop.mem library and storage size of the root partition only)

about_before2

After: (note the more accurate physical RAM size and storage size of the physical disk)

about_after2

### CODE REVIEW

**Comment by vjr** [comment]
I'm not sure how to fix the missing dependency error in the build stages... @elementary/desktop-developers

**Comment by vjr** [comment]
Here's what it looks like on my laptop - notice the disk size isn't rounded off but still acceptable IMO.

Before:

about_before

After:

about_after

**Comment by zeebok** [comment]
> I'm not sure how to fix the missing dependency error in the build stages... @elementary/desktop-developers

I believe you need to add those dependencies to the github workflow yml files

**Review by tintou — CHANGES_REQUESTED** [review]

File: src/Views/HardwareView.vala [hunk 1]
```diff
@@ -395,6 +415,27 @@ public class About.HardwareView : Gtk.Grid {
     private async void get_storage_info () {
         var file_root = GLib.File.new_for_path ("/");
         string storage_capacity = "";
+
+        uint64 storage_total = 0;
+
+        try {
+            UDisks.Client client = yield new UDisks.Client (null);
+            foreach (var object in client.object_manager.get_objects ()) {
```
[the remark this answers is not in the document]
Comment (vjr) [reply src/Views/HardwareView.vala]: Addressed, thank you!

**Review by tintou — APPROVED** [review]

**Pull request merged**